keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/handlers/GitDataGenerator.kt

```diff
@@ -0,0 +1,65 @@
+package com.netflix.spinnaker.keel.slack.handlers
+
+import com.netflix.spinnaker.keel.api.ScmInfo
+import com.netflix.spinnaker.keel.api.artifacts.GitMetadata
+import com.netflix.spinnaker.keel.api.artifacts.PublishedArtifact
+import com.netflix.spinnaker.keel.artifacts.getScmBaseLink
+import com.slack.api.model.kotlin_extension.block.SectionBlockBuilder
+import org.apache.logging.log4j.util.Strings
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+
+@Component
+class GitDataGenerator(
+  private val scmInfo: ScmInfo,
+  @Value("\${spinnaker.baseUrl}") private val spinnakerBaseUrl: String
+) {
+
+  fun generateRepoLink(gitMetadata: GitMetadata?): String {
```

Comment:
I find these functions with a single nullable parameter a bit weird. Since the function cannot do anything if the parameter is null, in my opinion it should not be nullable, and the caller should be responsible for checking that it has the data needed to make the call.

Reply:
I agree. Will think about it